packages/react/src/PricingOptions/PricingOptions.tsx

```diff
@@ -0,0 +1,479 @@
+import React, {forwardRef, HTMLAttributes, PropsWithChildren, type Ref} from 'react'
+import {CheckIcon, ChevronDownIcon, XIcon} from '@primer/octicons-react'
+import clsx from 'clsx'
+import type {BaseProps} from '../component-helpers'
+
+import {
+  Accordion,
+  Button,
+  ButtonBaseProps,
+  Heading,
+  HeadingProps,
+  Label,
+  Text,
+  UnorderedList,
+  UnorderedListProps,
+  useWindowSize,
+} from '..'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/pricing-cards/pricing-cards.css'
+
+/**
+ * Main stylesheet (as a CSS Module)
+ */
+import styles from './PricingOptions.module.css'
+
+export type PricingOptionsProps = {
+  presentation?: 'default' | 'cards'
+  ['data-testid']?: string
+} & PropsWithChildren<BaseProps<HTMLDivElement>>
+
+const testIds = {
+  root: 'PricingOptions',
+  item: 'PricingOptions__item',
+  heading: 'PricingOptions__heading',
+  label: 'PricingOptions__label',
+  description: 'PricingOptions__description',
+  price: 'PricingOptions__price',
+  primaryAction: 'PricingOptions__primaryAction',
+  secondaryAction: 'PricingOptions__secondaryAction',
+  featureList: 'PricingOptions__featureList',
+  featureListItem: 'PricingOptions__featureListItem',
+  featureListHeading: 'PricingOptions__featureListHeading',
+  footnote: 'PricingOptions__footnote',
+}
+
+const PricingOptionsRoot = forwardRef(
+  (
+    {
+      children,
+      className,
+      'data-testid': testId,
+      presentation = 'default',
+      ...rest
+    }: PropsWithChildren<PricingOptionsProps>,
+    ref: Ref<HTMLDivElement>,
+  ) => {
+    const filteredChildren = React.Children.toArray(children).filter(
+      child => React.isValidElement(child) && typeof child.type !== 'string' && child.type === PricingOptionsItem,
+    )
```

Comment:
Just a small note about memoization. May be worth it here given the value is very predictable, but expensive to produce each re-render.

```suggestion
    const filteredChildren = React.useMemo(React.Children.toArray(children).filter(
      child => React.isValidElement(child) && typeof child.type !== 'string' && child.type === PricingOptionsItem,
    ), [children])
```